### PR TITLE
OP-1199: Cameroon Voucher Scheme for subservices and subItems 

### PR DESCRIPTION
**Pull request by vaneck237**

### CODE REVIEW

**Comment by delcroip** [comment]
could you add the patch I attached so it will be compatible with BE changes
[claim_cs.txt](https://github.com/openimis/openimis-fe-claim_js/files/14402710/claim_cs.txt)

File: package.json [hunk 1]
```diff
@@ -23,7 +23,7 @@
     "@babel/cli": "^7.8.4",
     "@babel/core": "^7.9.6",
     "@babel/plugin-proposal-class-properties": "^7.8.3",
-    "@babel/plugin-transform-runtime": "^7.9.6",
+    "@babel/plugin-transform-runtime": "^7.19.1",
```
Comment (wzglinieckisoldevelo) [inline package.json:26]: Are you sure this should be changed? Is it needed? Also files like package-lock.json or yarn.lock should never be pushed to repo.

File: src/actions.js [hunk 2]
```diff
@@ -157,11 +157,39 @@ export function fetchClaimSummaries(mm, filters, withAttachmentsCount) {
 }
 
 export function formatDetail(type, detail) {
+  console.log(detail);
```
Comment (wzglinieckisoldevelo) [inline src/actions.js:160]: Remove all console.logs

**Review by ghost — CHANGES_REQUESTED** [review]

File: package-lock.json [hunk 3]
```diff
@@ -0,0 +1,5667 @@
+{
```
Comment (sniedzielski) [inline package-lock.json:1]: Do not push package-lock.json

File: src/actions.js [hunk 4]
```diff
@@ -176,9 +204,22 @@ export function formatDetail(type, detail) {
...
+    ${detail.qtyAsked !== null ? `qtyAsked: "${_.round(detail.qtyAsked, 2).toFixed(2) && _.round(detail.qtyDisplayed, 2).toFixed(2)}"` : ""}
+    ${detail.priceAsked !== null ? `priceAsked: "${_.round(detail.priceAsked, 2).toFixed(2)}"` : ""}
+    ${detail.qtyProvided !== null ? `qtyProvided: "${_.round(detail.qtyProvided, 2).toFixed(2)}"` : ""}
+  },`;
+}
+
 export function formatDetails(type, details) {
   if (!details) return "";
   let dets = details.filter((d) => !!d[type]);
+  console.log(`${type}s: [
```
Comment (wzglinieckisoldevelo) [inline src/actions.js:220]: Remove console.log

File: src/actions.js [hunk 5]
```diff
@@ -300,7 +341,10 @@ export function fetchClaim(mm, claimUuid, forFeedback) {
   } else {
     projections.push(
       "services{" +
-        "id, product { id, uuid }, service {id code name price maximumAmount} qtyProvided, priceAsked, qtyApproved, priceApproved, priceValuated, priceAdjusted, explanation, justification, rejectionReason, status" +
+        "id, product { id, uuid }, service {id code name price maximumAmount packagetype} qtyProvided,  priceAsked, qtyApproved, priceApproved, priceValuated,priceAdjusted, explanation, justification, rejectionReason, status," +
+        " claimlinkedItem{ item { id code name } qtyDisplayed priceAsked qtyProvided }"+
```
Comment (wzglinieckisoldevelo) [inline src/actions.js:345]: I would rewrite it as a separate projection to be concatenated, @olewandowski1 what do you think?

**Review by ghost — CHANGES_REQUESTED** [review]

File: src/pages/EditPage.js [hunk 6]
```diff
@@ -20,6 +20,15 @@ class EditPage extends Component {
 
 
   save = (claim) => {
+
+    this.claimPrefix = this.props.modulesManager.getConf(
+      "fe-claim",
+      "claimPrex",
+      0,
+    );
+    if(this.claimPrefix==1){
```
Comment (wzglinieckisoldevelo) [inline src/pages/EditPage.js:29]: If I understand this code correctly change this setting to false/true.

File: src/translations/en.json [hunk 7]
```diff
@@ -112,6 +112,7 @@
   "claim.edit.services.fetchingPricelist": " (fetching prices list…)",
   "claim.edit.services.totalClaimed": "Total Claimed: {totalClaimed}",
   "claim.edit.services.totalApproved": "Total Approved: {totalApproved}",
+  "claim.edit.services.MaxApproved": "Quantity set is more than quantity approved for this element. You will be reimbursed only for {totalApproved} element(s)",
```
Comment (wzglinieckisoldevelo) [inline src/translations/en.json:115]: @dragos-dobre I am not sure, but haven't we blocked this option already? You can't have more quantity than approved.

**Review by ghost — CHANGES_REQUESTED** [review]

File: yarn.lock [hunk 8]
```diff
@@ -0,0 +1,1861 @@
+# THIS IS AN AUTOGENERATED FILE. DO NOT EDIT THIS FILE DIRECTLY.
```
Comment (sniedzielski) [inline yarn.lock:1]: never push yarn.lock

**Pull request closed without merging**